front/components/assistant/conversation/UserMessage.tsx

```diff
@@ -37,6 +59,132 @@ export function UserMessage({
     []
   );
 
+  // We use this specific hook because this component is involved in the new conversation page.
+  const { agentConfigurations } = useUnifiedAgentConfigurations({
+    workspaceId: owner.sId,
+  });
+
+  const { featureFlags } = useFeatureFlags({ workspaceId: owner.sId });
+  const editMessagesFeatureFlag = featureFlags.includes("edit_messages");
+
+  const doEditMessage = useEditMessage(owner);
+
+  const router = useRouter();
+  async function switchThread(threadVersion: number | null) {
```

Comment:
`nit` To keep consistency with code below:
```suggestion
  const switchThread = async (threadVersion: number | null) => {
```